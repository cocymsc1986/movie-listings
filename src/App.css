

.app {
  padding: 1rem;
}

.filters__genres {
  display: flex;
  flex-flow: wrap;
}

.filters__genre {
  width: 25%;
  padding: 0.75rem;
  border: 1px solid #fff;
  background: #00bab3;
  color: white;
  text-align: center;
  box-sizing: border-box;
  cursor: pointer;
}

.filters__genre.filters__genre--selected {
  background: #019b95;
  box-shadow: inset 3px 3px 10px #018782;
}

.filters__genre label {
  pointer-events: none;
}

.filters__genre input[type=checkbox] {
  display: none;
}

.movies__results {
  display: flex;
  flex-flow: wrap;
}

.movies__result {
  width: 50%;
  display: flex;
  margin-bottom: 1rem;
}

.movies__image {
  margin-right: 0.5rem;
}

.movies__genre-labels {
  
}

.movies__genre-label {
  display: inline-block;
  padding: 0.25rem 0.5rem;
  margin-top: 0.5rem;
  margin-right: 0.5rem;
  font-size: 0.75rem;
  background: #5900bf;
  color: #fff;
}

.movies__genre-label:last-child {
  margin-right: 0;
}

@media only screen and (max-width: 720px) {
  .filters__genre {
    width: 50%;
  }

  .movies__result { 
    width: 100%;
  }
}