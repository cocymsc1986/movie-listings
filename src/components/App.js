import React, { Component } from 'react';

import Movie from './MovieContainer';
import Genres from './Genres';
import Rating from './Rating'

class App extends Component {
  render() {
    const {
      score_filter,
      updateScoreFilter,
      genres,
      movies,
      clickCheckBox,
      filterMovies,
      moviesLoading,
      genresLoading,
      genresError,
      moviesError
    } = this.props;

    return (
      <main className="app">
        <section className="filters">
          <h2>Filters</h2>
          <Rating 
            ratingValue={score_filter}
            updateScoreFilter={updateScoreFilter}
          />
          <h3>Genres</h3>
          {genresError ?
            'Error loading genre filters'
            :
            <div className="filters__genres">
              {genresLoading ?
                'Loading'
                :
                <Genres
                  genres={genres}
                  clickCheckBox={clickCheckBox}
                />
              }
            </div>
          }
        </section>
        <section class="movies"> 
          <h2>Movies</h2>
          {moviesError ?
            'Error loading movies'
            :
            <div className="movies__results">
              {moviesLoading ?
                'Loading'
                :
                movies && filterMovies(movies).map((movie, key) => {
                  return (
                    <Movie
                      imagePath={movie.poster_path}
                      title={movie.title}
                      genreIds={movie.genre_ids}
                      genres={genres}
                      key={key}
                    />
                  )
                })
              }
            </div>
          }
        </section>
      </main>
    );
  }
}

export default App;
