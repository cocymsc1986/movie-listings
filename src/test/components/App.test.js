import React from 'react';

import App from '../../components/App';

describe('App component', () => {

	test('Should render', () => {
		const wrapper = shallow(<App />);
		expect(wrapper).toMatchSnapshot();
	});

	test('Should render with moviesError', () => {
		const wrapper = shallow(<App moviesError={true} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('Should render with genresError', () => {
		const wrapper = shallow(<App genresError={true}/>);
		expect(wrapper).toMatchSnapshot();
	});
});