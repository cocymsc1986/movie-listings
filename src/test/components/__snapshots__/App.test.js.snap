// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="filters"
>
          <h2>
                    Filters
          </h2>
          <Unknown
                    ratingValue={undefined}
                    updateScoreFilter={undefined}
          />
          <h3>
                    Genres
          </h3>
          <div
                    className="filters__genres"
          >
                    <Genres
                              clickCheckBox={undefined}
                              genres={undefined}
                    />
          </div>
</section>,
        <section
          class="movies"
>
          <h2>
                    Movies
          </h2>
          <div
                    className="movies__results"
          />
</section>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Filters
</h2>,
            <Unknown
              ratingValue={undefined}
              updateScoreFilter={undefined}
/>,
            <h3>
              Genres
</h3>,
            <div
              className="filters__genres"
>
              <Genres
                            clickCheckBox={undefined}
                            genres={undefined}
              />
</div>,
          ],
          "className": "filters",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Filters",
            },
            "ref": null,
            "rendered": "Filters",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "ratingValue": undefined,
              "updateScoreFilter": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Genres",
            },
            "ref": null,
            "rendered": "Genres",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Genres
                clickCheckBox={undefined}
                genres={undefined}
/>,
              "className": "filters__genres",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "clickCheckBox": undefined,
                "genres": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Movies
</h2>,
            <div
              className="movies__results"
/>,
          ],
          "class": "movies",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movies",
            },
            "ref": null,
            "rendered": "Movies",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "movies__results",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="filters"
>
            <h2>
                        Filters
            </h2>
            <Unknown
                        ratingValue={undefined}
                        updateScoreFilter={undefined}
            />
            <h3>
                        Genres
            </h3>
            <div
                        className="filters__genres"
            >
                        <Genres
                                    clickCheckBox={undefined}
                                    genres={undefined}
                        />
            </div>
</section>,
          <section
            class="movies"
>
            <h2>
                        Movies
            </h2>
            <div
                        className="movies__results"
            />
</section>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Filters
</h2>,
              <Unknown
                ratingValue={undefined}
                updateScoreFilter={undefined}
/>,
              <h3>
                Genres
</h3>,
              <div
                className="filters__genres"
>
                <Genres
                                clickCheckBox={undefined}
                                genres={undefined}
                />
</div>,
            ],
            "className": "filters",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filters",
              },
              "ref": null,
              "rendered": "Filters",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "ratingValue": undefined,
                "updateScoreFilter": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Genres",
              },
              "ref": null,
              "rendered": "Genres",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Genres
                  clickCheckBox={undefined}
                  genres={undefined}
/>,
                "className": "filters__genres",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "clickCheckBox": undefined,
                  "genres": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Movies
</h2>,
              <div
                className="movies__results"
/>,
            ],
            "class": "movies",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movies",
              },
              "ref": null,
              "rendered": "Movies",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "movies__results",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`App component Should render with genresError 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    genresError={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="filters"
>
          <h2>
                    Filters
          </h2>
          <Unknown
                    ratingValue={undefined}
                    updateScoreFilter={undefined}
          />
          <h3>
                    Genres
          </h3>
          Error loading genre filters
</section>,
        <section
          class="movies"
>
          <h2>
                    Movies
          </h2>
          <div
                    className="movies__results"
          />
</section>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Filters
</h2>,
            <Unknown
              ratingValue={undefined}
              updateScoreFilter={undefined}
/>,
            <h3>
              Genres
</h3>,
            "Error loading genre filters",
          ],
          "className": "filters",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Filters",
            },
            "ref": null,
            "rendered": "Filters",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "ratingValue": undefined,
              "updateScoreFilter": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Genres",
            },
            "ref": null,
            "rendered": "Genres",
            "type": "h3",
          },
          "Error loading genre filters",
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Movies
</h2>,
            <div
              className="movies__results"
/>,
          ],
          "class": "movies",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movies",
            },
            "ref": null,
            "rendered": "Movies",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "movies__results",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="filters"
>
            <h2>
                        Filters
            </h2>
            <Unknown
                        ratingValue={undefined}
                        updateScoreFilter={undefined}
            />
            <h3>
                        Genres
            </h3>
            Error loading genre filters
</section>,
          <section
            class="movies"
>
            <h2>
                        Movies
            </h2>
            <div
                        className="movies__results"
            />
</section>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Filters
</h2>,
              <Unknown
                ratingValue={undefined}
                updateScoreFilter={undefined}
/>,
              <h3>
                Genres
</h3>,
              "Error loading genre filters",
            ],
            "className": "filters",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filters",
              },
              "ref": null,
              "rendered": "Filters",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "ratingValue": undefined,
                "updateScoreFilter": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Genres",
              },
              "ref": null,
              "rendered": "Genres",
              "type": "h3",
            },
            "Error loading genre filters",
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Movies
</h2>,
              <div
                className="movies__results"
/>,
            ],
            "class": "movies",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movies",
              },
              "ref": null,
              "rendered": "Movies",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "movies__results",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`App component Should render with moviesError 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    moviesError={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="filters"
>
          <h2>
                    Filters
          </h2>
          <Unknown
                    ratingValue={undefined}
                    updateScoreFilter={undefined}
          />
          <h3>
                    Genres
          </h3>
          <div
                    className="filters__genres"
          >
                    <Genres
                              clickCheckBox={undefined}
                              genres={undefined}
                    />
          </div>
</section>,
        <section
          class="movies"
>
          <h2>
                    Movies
          </h2>
          Error loading movies
</section>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Filters
</h2>,
            <Unknown
              ratingValue={undefined}
              updateScoreFilter={undefined}
/>,
            <h3>
              Genres
</h3>,
            <div
              className="filters__genres"
>
              <Genres
                            clickCheckBox={undefined}
                            genres={undefined}
              />
</div>,
          ],
          "className": "filters",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Filters",
            },
            "ref": null,
            "rendered": "Filters",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "ratingValue": undefined,
              "updateScoreFilter": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Genres",
            },
            "ref": null,
            "rendered": "Genres",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Genres
                clickCheckBox={undefined}
                genres={undefined}
/>,
              "className": "filters__genres",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "clickCheckBox": undefined,
                "genres": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Movies
</h2>,
            "Error loading movies",
          ],
          "class": "movies",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Movies",
            },
            "ref": null,
            "rendered": "Movies",
            "type": "h2",
          },
          "Error loading movies",
        ],
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="filters"
>
            <h2>
                        Filters
            </h2>
            <Unknown
                        ratingValue={undefined}
                        updateScoreFilter={undefined}
            />
            <h3>
                        Genres
            </h3>
            <div
                        className="filters__genres"
            >
                        <Genres
                                    clickCheckBox={undefined}
                                    genres={undefined}
                        />
            </div>
</section>,
          <section
            class="movies"
>
            <h2>
                        Movies
            </h2>
            Error loading movies
</section>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Filters
</h2>,
              <Unknown
                ratingValue={undefined}
                updateScoreFilter={undefined}
/>,
              <h3>
                Genres
</h3>,
              <div
                className="filters__genres"
>
                <Genres
                                clickCheckBox={undefined}
                                genres={undefined}
                />
</div>,
            ],
            "className": "filters",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filters",
              },
              "ref": null,
              "rendered": "Filters",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "ratingValue": undefined,
                "updateScoreFilter": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Genres",
              },
              "ref": null,
              "rendered": "Genres",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Genres
                  clickCheckBox={undefined}
                  genres={undefined}
/>,
                "className": "filters__genres",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "clickCheckBox": undefined,
                  "genres": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Movies
</h2>,
              "Error loading movies",
            ],
            "class": "movies",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Movies",
              },
              "ref": null,
              "rendered": "Movies",
              "type": "h2",
            },
            "Error loading movies",
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
